store/src/db/mod.rs

```diff
@@ -52,6 +53,14 @@ pub struct NoteCreated {
     pub tag: u64,
 }
 
+impl NoteCreated {
+    /// Returns the absolute position on the note tree based on the batch index
+    /// and local-to-the-subtree index
+    pub fn absolute_note_index(&self) -> u32 {
+        BlockNoteTree::leaf_index(self.batch_index as usize, self.note_index as usize) as u32
+    }
```

Comment:
I wonder if this should be:
```Rust
BlockNoteTree::leaf_index(self.batch_index as usize, self.note_index as usize) as u32 / 2
```

This would give us the index of the node which is the parent of both note leaf and metadata leaf, and I think that maybe this is what the client is expecting?

Reply:
That very much sounds like it. I tested it and now I don't get [the transaction executor's `is_in_block()` error](https://github.com/0xPolygonMiden/miden-base/blob/69ec4dc4e1a6f78d3d370c6b3b2bd25b5887ebc8/objects/src/transaction/inputs.rs#L84-L88). I get a VM assertion error

```rust
called `Result::unwrap()` on an `Err` value: TransactionExecutionError(ExecuteTransactionProgramFailed(FailedAssertion { clk: 5456, err_code: 131075, err_msg: None }))
```